common/src/View/IssueBrowserView.cpp

```diff
@@ -154,14 +155,17 @@ namespace TrenchBroom {
         }
 
         std::vector<Model::Issue*> IssueBrowserView::collectIssues(const QList<QModelIndex>& indices) const {
-            std::vector<Model::Issue*> result;
+            // Use a vector_set to filter out duplicates.
+            // The QModelIndex list returned by getSelection() contains duplicates
+            // (not sure why, current row and selected row?)
+            kdl::vector_set<Model::Issue*> result;
```

Comment:
Can you reserve enough space for the vector to avoid reallocation?

Reply:
Will do!

Reply:
Btw, the `vector_set` constructors taking a capacity have a dangerous asymmetry with std::vector, where std::vector(size_t n) fills the vector with `n` default initialized objects, rather than producing a size 0 container with storage reserved.

I never use container constructors that takes a size for that reason, also it's not clear what `std::vector<size_t> foo {2}` is ({2} or {0, 0}).  I'd suggest we just drop them from vector_set.

Reply:
Ok, we can do that. But I'd like to retain some way to reserve space beforehand, so maybe we need to add a function that does this.

Reply:
I added `set_adapter::reserve()`, that should be good enough right?
My understanding is:
```
std::vector<int> foo;
foo.reserve(300);
```
is optimal, at least on MSVC the first line doesn't allocate any storage,

Reply:
I think that function should go into `vector_set`, because we cannot be sure that the underlying collection of `vector_set` has a `reserve` member function. For example, you can wrap a `vector_set' around a `std::list`, too.

Yeah, I'd be surprised if there's any allocation until the first push_back / emplace or whatever.

Reply:
Makes sense, I put in there because set_adapter was already using `m_data.reserve`. But anyway I'll make another issue for these changes

Reply:
Oh yeah, looks like there are two misplaced calls there.